src/assets/_project/_blocks/utils/qg-misc.js

```diff
@@ -70,10 +70,14 @@
       label: 'Alert',
     },
   ];
-  deprecationWarnings.forEach(({selector, label}) => {
-    // use `show` if the element is created on the fly
-    $(selector).show(() => {
-      console.warn(`Please change the font awesome element in ${label} from i to span, we'll be removing the css in this element before 22nd june 2022. Please refer to the https://github.com/qld-gov-au/qg-web-template/pull/391 for more details.`);
-    });
-  });
+  setTimeout(
+    () => {
+      deprecationWarnings.forEach(({selector, label}) => {
+        if ($(selector).length) {
+          console.warn(`Please change the font awesome element in ${label} from i to span, we'll be removing the css in this element before 22nd june 2022. Please refer to the https://github.com/qld-gov-au/qg-web-template/pull/391 for more details.`);
+        }
+      });
+    },
+    1000,
```

Comment:
set timeout with some defined time is not stable (1000) because there can be times when a page takes more time to load etc.

If you just want to use to wait for the event loop then I would suggest to use '0' as timeout or go for 'promise'

Reply:
tested on uat (https://oss-uat.clients.squiz.net/health/staying-healthy/food-pantry/food-labelling/label-buster/_nocache) 0 doesn't work but 1000 is working, i've commented it as a temporary solution for SPA

other solutions to detect dom creations (https://stackoverflow.com/questions/3384398/how-to-detect-new-element-creation-in-jquery) seems a bit too aggressive and involve plugin which is a bit risky to take for such a task

Reply:
this should catch most, i'd suggest 5 second wait or even 10, as this is about casual notification and does not need to be blocking the main processing thread.